docs/design/unified-design.md

```diff
@@ -61,86 +85,139 @@ sequenceDiagram
+- **Worker Customization:**  
+  Users can implement custom `WorkerLoad` classes to handle specialized tasks such as data preprocessing, reinforcement
+  learning, or other domain-specific operations. This flexibility allows seamless integration of varied processing
+  strategies within the hybrid training pipeline.
 
-- **Worker Customization:** Users can implement custom `WorkerLoad` classes to handle specific workloads, such as data preprocessing or reinforcement learning tasks. This allows for flexible integration of different processing strategies within the hybrid training framework.
-- **Custom SubMaster:** Custom `SubMaster` implementations can be created to manage specific training processes, enabling tailored orchestration of elastic workloads.
+- **Custom Backend:**  
+  Users may define their own `SubMaster` and corresponding `Worker` implementations to orchestrate specific training
+  workflows. This supports fine-grained control over elastic workload management and enables backend customization for
+  specialized use cases.
 
-The `Worker/SubMaster` is driven by `PrimeManager` through lifecycle hooks, allowing it to manage the lifecycle of each workload. This design enables users to easily extend the framework to support new processing paradigms without modifying the core architecture.
+The `Worker` and `SubMaster` components are orchestrated by the `PrimeManager` via stages and lifecycle hooks, which govern the
+initialization, execution, and monitoring of each workload. This hook-based design empowers users to extend the
+framework with new processing paradigms—without altering the core architecture.
 
-Common Hooks include:
+### Common Lifecycle Hooks
 
-- `__init__`: Initialize the node with configuration.
-- `status`: Get the running status of the node.
-- `self_check`: Perform self-checks to ensure the node is healthy.
-- `check_child`: Check the status of child nodes (for SubMaster).
-- `start`: Start the node's processing. (e.g., start training loop or monitoring)
+- `__init__`: Initializes the node with the provided configuration.
+- `status`: Retrieves the current runtime status of the node.
+- `check_child`: Monitors the status of child nodes (applicable to `SubMaster`).
+- `start`: Launches the node’s processing logic (e.g., training loop, monitoring routine).
+
+### Core Lifecycle Stages
+
+- `INIT`: The initial stage of the actor lifecycle.
+- `READY`: The actor has completed its setup and is ready to handle RPC requests.
+- `RUNNING`: Indicates that the task is actively running; set after the `start` hook is invoked.
+- `FINISH` / `FAILED`: Terminal stages that signify the completion or failure of the task.
+
+```mermaid
+stateDiagram-v2
+    [*] --> INIT: __init__
+    INIT --> READY: _setup, _self_check, ...
+    READY --> RUNNING: check_workers, rendezvous, ...
+    RUNNING --> FINISH: 
+    RUNNING --> FAILED: task error
+    FINISH --> [*]
+    FAILED --> [*]
+```
 
 ### 稳定性
 
 #### Pre check
```

Comment:
We can give some description here for the current default node check, and leave a TODO for other pre-checks that can be extended in the future.